core/lib/state/src/mempool_cache/mod.rs

```diff
@@ -0,0 +1,81 @@
+mod metrics;
+
+use std::{future::Future, sync::Arc, time::Duration};
+
+use chrono::NaiveDateTime;
+use tokio::sync::{watch, RwLock};
+use zksync_dal::ConnectionPool;
+use zksync_types::H256;
+
+use crate::{cache::sequential_cache::SequentialCache, mempool_cache::metrics::METRICS};
+
+/// Used for `eth_getPendingTransactions` requests on API servers
```

Comment:
nit: there is no `eth_getPendingTransactions`. It's a `eth_newPendingTransactionFilter` + `eth_getFilterChanges`, but mentioning only latter is fine